pkg/cmd/openshift-tests/collect-disk-certificates/collector.go

```diff
@@ -0,0 +1,217 @@
+		`),
+
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancelFn := context.WithCancel(context.Background())
+			defer cancelFn()
+			abortCh := make(chan os.Signal, 2)
+			go func() {
+				<-abortCh
+				fmt.Fprintf(f.ErrOut, "Interrupted, terminating\n")
+				cancelFn()
+
+				sig := <-abortCh
+				fmt.Fprintf(f.ErrOut, "Interrupted twice, exiting (%s)\n", sig)
+				switch sig {
+				case syscall.SIGINT:
+					os.Exit(130)
+				default:
+					os.Exit(0)
+				}
+			}()
+			signal.Notify(abortCh, syscall.SIGINT, syscall.SIGTERM)
+
+			if err := f.Validate(); err != nil {
+				return err
+			}
+
+			o, err := f.ToOptions()
+			if err != nil {
+				return err
+			}
+
+			return o.Run(ctx)
+		},
+	}
+
+	f.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (f *RunCollectDiskCertificatesFlags) AddFlags(flags *pflag.FlagSet) {
+	flags.StringArrayVar(&f.CollectDirs, "collect-dir", f.CollectDirs, "directories to collect certs in")
+	flags.StringVar(&f.Prefix, "prefix", f.Prefix, "directories prefix")
+	f.ConfigFlags.AddFlags(flags)
+	f.OutputFlags.BindFlags(flags)
+}
+
+func (f *RunCollectDiskCertificatesFlags) SetIOStreams(streams genericclioptions.IOStreams) {
+	f.IOStreams = streams
+}
+
+func (f *RunCollectDiskCertificatesFlags) Validate() error {
+	if len(f.OutputFlags.OutFile) == 0 {
+		return fmt.Errorf("output-file must be specified")
+	}
+	if len(f.CollectDirs) == 0 {
+		return fmt.Errorf("dirs must be specified")
+	}
+
+	return nil
+}
+
+func (f *RunCollectDiskCertificatesFlags) ToOptions() (*RunCollectDiskCertificatesOptions, error) {
+	originalOutStream := f.IOStreams.Out
+	closeFn, err := f.OutputFlags.ConfigureIOStreams(f.IOStreams, f)
+	if err != nil {
+		return nil, err
+	}
+
+	restConfig, err := f.ConfigFlags.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RunCollectDiskCertificatesOptions{
+		KubeClient:      kubeClient,
+		OutputFile:      f.OutputFlags.OutFile,
+		CloseFn:         closeFn,
+		OriginalOutFile: originalOutStream,
+		IOStreams:       f.IOStreams,
+		CollectDirs:     f.CollectDirs,
+		Prefix:          f.Prefix,
+	}, nil
+}
+
+// RunCollectDiskCertificatesOptions sets options for api server disruption monitor
+type RunCollectDiskCertificatesOptions struct {
+	KubeClient kubernetes.Interface
+	OutputFile string
+
+	CollectDirs []string
+	Prefix      string
+
+	OriginalOutFile io.Writer
+	CloseFn         iooptions.CloseFunc
+	genericclioptions.IOStreams
+}
+
+func (o *RunCollectDiskCertificatesOptions) Run(ctx context.Context) error {
+	ctx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+
+	pkiList := &certgraphapi.PKIList{}
+	errs := []error{}
+
+	restConfig, err := clusterinfo.GetMonitorRESTConfig()
+	if err != nil {
+		return err
+	}
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
```

Comment:
`RunCollectDiskCertificatesOptions` already has a kubeClient, why create another one instead of `o.KubeClient`?